pkg/skaffold/runner/dev.go

```diff
@@ -119,6 +120,19 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *kuber
 		}
 	}
 
+	if needsTest {
+		event.ResetStateOnTest()
+		defer func() {
+			r.changeSet.resetTest()
+		}()
+
+		if err := r.Test(ctx, out, r.builds); err != nil {
+			logrus.Warnln("Skipping deploy due to test error:", err)
```

Comment:
This log message should be conditional on `needsDeploy`. Please fix above also.

Reply:
Added `needsDeploy` condition for the warning.